arm-operationalinsights/2015-03-20/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1110 @@
+        }
+      },
+      "description": "Parameters specifying the search query and range."
+    },
+    "StorageAccount": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The Azure Resource Manager ID of the storage account resource."
+        },
+        "key": {
+          "type": "string",
+          "description": "The storage account key."
+        }
+      },
+      "description": "Describes a storage account connection."
+    },
+    "StorageInsightStatus": {
+      "properties": {
+        "state": {
+          "type": "string",
+          "description": "The state of the storage insight connection to the workspace"
+        }
+      },
+      "description": "The status of the storage insight."
+    },
+    "StorageInsightProperties": {
+      "properties": {
+        "containers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The names of the blob containers that the workspace should read"
+        },
+        "tables": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The names of the Azure tables that the workspace should read"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccount",
+          "description": "The storage account connection details"
+        },
+        "status": {
+          "$ref": "#/definitions/StorageInsightStatus",
+          "description": "The status of the storage insight"
+        }
+      },
+      "required": [
+        "storageAccount"
+      ],
+      "description": "Storage insight properties."
+    },
+    "StorageInsight": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/StorageInsightProperties",
+          "description": "Storage insight properties."
+        },
+        "eTag": {
+          "type": "string",
+          "description": "The ETag of the storage insight."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "The top level storage insight resource container."
+    },
+    "StorageInsightListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageInsight"
+          },
+          "description": "Gets or sets a list of storage insight instances."
+        },
+        "@odata.nextLink": {
+          "type": "string",
+          "description": "The link (url) to the next page of results."
+        }
+      },
+      "description": "The list storage insights operation response."
+    },
+    "Resource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
```

Comment:
`CompositeTypes are not the same.` is correct, since both versions of `Resource` (`2015-03-20` and `2015-11-01-preview`) have to match up. In other words, make the change in both files.